src/Monolog/Formatter/JsonFormatter.php

```diff
@@ -68,13 +70,24 @@ public function isAppendingNewlines(): bool
     public function format(array $record): string
     {
         $normalized = $this->normalize($record);
+
         if (isset($normalized['context']) && $normalized['context'] === []) {
             $normalized['context'] = new \stdClass;
         }
         if (isset($normalized['extra']) && $normalized['extra'] === []) {
             $normalized['extra'] = new \stdClass;
         }
 
+        if ($this->ignoreEmptyContextAndExtra) {
+            if ((array) $normalized['context'] === []) {
```

Comment:
I think it might make more sense to add a `if $(this->ignoreEmptyContextAndExtra)` check above at line 75/78 to do unset in that case instead of creating a stdClass and then casting it to array here and unsetting it again.. That seems a little wasteful.